### ISSUE TITLE
Periodically retrain models on real data, verify performance.

### ISSUE THREAD

**Issue by brendan-ai2** [allenai/allennlp#1813]
**Is your feature request related to a problem? Please describe.**
Though the library has good unit test coverage, we don't automatically train our models on real data. This leaves room for regressions and potential blind spots. For instance, https://github.com/allenai/allennlp/issues/1134, where the validation loss for a particular model was atrocious.

**Describe the solution you'd like**
Periodically we should retrain our models on real data and verify that they give acceptable performance on the validation set. This would be a TeamCity job (or jobs) running perhaps nightly or weekly to catch regressions.

**Describe alternatives you've considered**
1. We could try to write "better" unit tests, but this is not actionable in an obvious way.
2. We could have these retraining tests run as unit tests. This would be too slow.

**Additional context**
Bit of discussion on: https://github.com/allenai/allennlp/pull/1808

**Comment by schmmd** [comment]
Closing this out, now that we have model-ci.allennlp.org (although it's still WIP).
